spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -560,20 +560,15 @@ public Scan buildChangelogScan() {
     }
 
     boolean emptyScan = false;
-    if (startTimestamp != null) {
-      if (table.currentSnapshot() != null
-          && table.currentSnapshot().timestampMillis() < startTimestamp) {
-        emptyScan = true;
+    if (startTimestamp != null || endTimestamp != null) {
+      if (startTimestamp != null) {
+        startSnapshotId = getStartSnapshotId(startTimestamp);
       }
-      startSnapshotId = getStartSnapshotId(startTimestamp);
-      if (startSnapshotId == null && endTimestamp == null) {
-        emptyScan = true;
+      if (endTimestamp != null) {
+        endSnapshotId = getEndSnapshotId(startTimestamp, endTimestamp);
       }
-    }
-
-    if (endTimestamp != null) {
-      endSnapshotId = getEndSnapshotId(endTimestamp);
       if ((startSnapshotId == null && endSnapshotId == null)
+          || (endSnapshotId == null && endTimestamp != null)
           || (startSnapshotId != null && startSnapshotId.equals(endSnapshotId))) {
         emptyScan = true;
       }
```

Comment:
I believe this part of logic should be out side of line 563 ` if (startTimestamp != null || endTimestamp != null) `, as we are not only check the when timestamp is one of the inputs, we also need to check when inputs are start snapshot id and end snapshot id.